drivers/net/oa_tc6/oa_tc6.c

```diff
@@ -0,0 +1,2176 @@
+
+          /* Set index so that a subsequent chunk with
+           * smaller payload won't pass
+           */
+
+          priv->rx_pkt_idx = OA_TC6_MAX_FRAME_SIZE(priv) + 1;
+          return;
+        }
+
+      netpkt_copyin(&priv->dev, priv->rx_pkt, rxbuf,
+                    rxlen, priv->rx_pkt_idx);
+      priv->rx_pkt_idx = newlen;
+
+      if (oa_tc6_end_valid(footer))
+        {
+          oa_tc6_finalize_rx_packet(priv);
+        }
+    }
+}
+
+/****************************************************************************
+ * Name: oa_tc6_finalize_rx_packet
+ *
+ * Description:
+ *   Strip down last 4 bytes (FCS) from the rx packet, mark it ready
+ *   and notify upper.
+ *
+ * Input Parameters:
+ *   priv - pointer to the driver-specific state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void oa_tc6_finalize_rx_packet(FAR struct oa_tc6_driver_s *priv)
+{
+  netpkt_setdatalen(&priv->dev, priv->rx_pkt,
+                    netpkt_getdatalen(&priv->dev, priv->rx_pkt) - 4);
+  priv->rx_pkt_ready = true;
+  netdev_lower_rxready(&priv->dev);
+}
+
+/****************************************************************************
+ * Name: oa_tc6_release_tx_packet
+ *
+ * Description:
+ *   Release the tx packet.
+ *
+ * Input Parameters:
+ *   priv - pointer to the driver-specific state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void oa_tc6_release_tx_packet(FAR struct oa_tc6_driver_s *priv)
+{
+  netpkt_free(&priv->dev, priv->tx_pkt, NETPKT_TX);
+  priv->tx_pkt = NULL;
+}
+
+/****************************************************************************
+ * Name: oa_tc6_release_rx_packet
+ *
+ * Description:
+ *   Release the rx packet.
+ *
+ * Input Parameters:
+ *   priv - pointer to the driver-specific state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void oa_tc6_release_rx_packet(FAR struct oa_tc6_driver_s *priv)
+{
+  netpkt_free(&priv->dev, priv->rx_pkt, NETPKT_RX);
+  priv->rx_pkt = NULL;
+}
+
+/****************************************************************************
+ * Name: oa_tc6_(select/deselect)_spi
+ *
+ * Description:
+ *   Helper functions to setup SPI hardware.
+ *
+ * Input Parameters:
+ *   priv - pointer to the driver-specific state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void oa_tc6_select_spi(FAR struct oa_tc6_driver_s *priv)
+{
+  SPI_LOCK(priv->spi, true);
+
+  SPI_SETMODE(priv->spi, OA_TC6_SPI_MODE);
+  SPI_SETBITS(priv->spi, OA_TC6_SPI_NBITS);
+  SPI_HWFEATURES(priv->spi, 0);  /* disable HW features */
+  SPI_SETFREQUENCY(priv->spi, priv->config->frequency);
+#ifdef CONFIG_SPI_DELAY_CONTROL
+  SPI_SETDELAY(priv->spi, 0, 0, 0, 0);
```

Comment:
Delay values should be configurable, likely from Kconfig. You can check `lcd/st7789.c` driver. But this is a nitpick, can be added in a later patch if you want.

Reply:
I expect that OA compliant chips should need no dělay at given specified bitrate up to the specified limit. There could be required delay in chipselect if there is problem with unsynchronized paths to GIO chip-selects and SPI pins. But it would be rare, if chip select is driven by SPI, it has defined timing according to mode 0 to 3 and it should be enough. If it is GPIO chip select then delay by executing even few instructions and access over buses is enough as well. May it be, possible requirements for delay can be specified by low level specific chip driver if it has such requirement.

So I would not introduce that until there is some chip or SPI peripheral combination which has problems.